qiita_db/util.py

```diff
@@ -1391,11 +1391,12 @@ def generate_study_list(study_ids, build_samples):
 
             # shared with
             info['shared'] = []
-            for name, email in zip(info['shared_with_name'],
-                                   info['shared_with_email']):
-                if not name:
-                    name = email
-                info['shared'].append((email, name))
+            if info['shared_with_name'] and info['shared_with_email']:
```

Comment:
It looks like these lines of code where bugs introduced in the previous PR, which means that the tests do not capture this case. I recommend to add a specific test that will capture this case so it doesn't repeat.